src/tools/tests/manifest2proto-test.ts

```diff
@@ -777,7 +786,25 @@ describe('manifest2proto', () => {
 
     assert.deepStrictEqual(varInput, varOutput);
     assert.deepStrictEqual(varInput.name, 'a');
-    assert.isUndefined(varInput.constraint);
+    assert.isFalse(varInput.constraint.unconstrained);
+    assert.isFalse(varOutput.constraint.unconstrained);
+  });
+
+  it('encodes max variable type for particle specs - unconstrained', async () => {
```

Comment:
Can we add the following test where `unconstrained` should be false and `constraintType` is empty (or null).

```
Particle P
   input: reads ~a 
   output: writes ~a
```